core/src/main/java/org/aerogear/mobile/core/http/OkHttpRequest.java

```diff
@@ -27,54 +31,116 @@ public OkHttpRequest(final OkHttpClient client, final AppExecutors executors) {
         this.headers.put(CONTENT_TYPE_HEADER, JSON_MIME_TYPE);
     }
 
-    @Override
+
     public HttpRequest addHeader(final String key, final String value) {
         this.headers.put(key, value);
         return this;
     }
 
     @Override
-    public void get(final String url) {
-        Request.Builder getRequestBuilder = requestBuilderWithUrl(url);
-        addHeaders(getRequestBuilder);
-        Request getRequest = getRequestBuilder.build();
-        call = client.newCall(getRequest);
-    }
+    public org.aerogear.mobile.core.reactive.Request<HttpResponse> get(final String url) {
+        AtomicReference<OkHttpResponse> response = new AtomicReference<>();
+        return Requester.call(() -> {
+            Builder getRequestBuilder = requestBuilderWithUrl(url);
+            addHeaders(getRequestBuilder);
+            Request getRequest = getRequestBuilder.build();
+            call = client.newCall(getRequest);
+
+            OkHttpResponse okHttpResponse = new OkHttpResponse(call);
+            response.set(okHttpResponse);
+            return (HttpResponse) okHttpResponse;// cast to force generic type
+        }, () -> {
+            /* Cleaner */
+            OkHttpResponse okHttpResponse = response.get();
+            if (okHttpResponse != null && !okHttpResponse.isClosed()) {
+                okHttpResponse.waitForCompletionAndClose();
+            }
+        }).cancelWith(() -> {
+            call.cancel();
+        }).requestOn(appExecutors.networkThread());
```

Comment:
so looks like all the requests will be executed on the network thread, and it is a single thread. If a request is taking too long to execute, all the other requests will be blocked. I think that is a concern. Should we make the number of threads configurable for the network request executor?

Reply:
Yes, can you make a github issue?